### DISCUSSION TITLE
Get All Alerts

### DISCUSSION THREAD

**Discussion by CoryatJohn** [weather-gov/api#717] [Q&A] [answered]
I would like to pull all the alerts in a single query to populate the alert system in our application. Right now, I see around 350 active alerts and there's little weather. I can see alert counts going above 500. I tried using the limit=1000 parameter and it was rejected. The documentation says 500 is the default. The error code says 500 is the maximum.

So how do I pull the entire list of alerts? It's not an issue if it takes several requests but it does make it more complicated.

Thanks!

(RadarNow!)

Edit: I may have found my own answer. Use a start time. I think this will solve it nicely.

Decoded: (do not click)
https://api.weather.gov/alerts?start=2024-04-20T15:00:00+00:00&status=actual

Encoded:
https://api.weather.gov/alerts?start=2024-04-20T15%3A00%3A00%2B00%3A00&status=actual

**Comment by netbymatt** [comment, marked as answer]
Set the limit really low. You'll get a "pagination" property in the response. Follow the provided link.

Here's an excerpt from https://api.weather.gov/alerts?limit=5

```
  "pagination": {
    "next": "https://api.weather.gov/alerts?limit=5&cursor=eyJ0IjoxNzEzNjQ1MDAwLCJpIjoidXJuOm9pZDoyLjQ5LjAuMS44NDAuMC5kNWZjOThmYzU5ZmFmODcyZWZiMWJjMDg4YTM0NjllYjI4MmFlYzZhLjAwMS4xIn0%3D"
}
```

**Reply by CoryatJohn** [reply]
Perfect! Thanks!

**Comment by sullynole** [comment]
Currently the limit applies to /alerts with a max of 500 (default).  /alerts/active does not currently use the limit feature (it will return all active alerts even if more than 500 and the limit filter can't be used either).

**Reply by CoryatJohn** [reply]
When I request the alert list without a limit, I get 500 alerts and no pagination cursor.

**Reply by scadergit** [reply]
As @sullynole noted, the full path to active alerts /alerts/active does not have a limit and will return all active alerts in one request. Using just /alerts with the parameter "active" might honor the limit request, but that would require multiple requests. Using the full path is recommended and supports the other parameters.

**Reply by CoryatJohn** [reply]
The URL: https://api.weather.gov/alerts returns exactly 500 alerts. It also has a pagination parameter, so I was incorrect in stating that.

Looks like the maximum is 500 no matter what. So using pagination along with a start date will certainly solve my issue.

**Reply by scadergit** [reply]
Are you looking for all alerts, even expired alerts, or just active alerts?

**Reply by CoryatJohn** [reply]
I have my answer. This thread can be closed.
